python_modules/dagster/dagster_tests/core_tests/test_solid_invocation.py

```diff
@@ -693,3 +694,18 @@ async def my_coroutine_test():
 
     loop = asyncio.get_event_loop()
     loop.run_until_complete(my_coroutine_test())
+
+
+def test_solid_invocation_nothing_deps():
+    @solid(input_defs=[InputDefinition("start", Nothing)])
+    def install_deps():
+        return 5
+
+    # Ensure that providing the Nothing-dependency input works
+    assert install_deps(start="blah") == 5
+
+    # Ensure that if not provided, Dagster throws that input is expected.
+    with pytest.raises(
+        DagsterInvalidInvocationError, match='No value provided for required input "start"'
+    ):
+        install_deps()
```

Comment:
I don't think this makes sense. We should at minimum allow you to execute it without passing anything, and potentially just not allow you to pass anything (but have a good error message). 

I think one framing for direct invocation is that its a way to execute the decorated function as directly as possible. In this case, the decorated function takes no arguments, so I should execute it by passing no arguments. 

Another framing thing is that `Nothing` is distinct from `None` in that `None` is a value and `Nothing` is the absence of any value.